### PR TITLE
codelyzer - update codelyzer configs

### PR DESCRIPTION
**Pull request by SuperITMan**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/code-style/blob/master/CONTRIBUTING.md#-commit-message-guidelines
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [X] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Code-Style does not contain configurations for `codelyzer` 5.0.0 to 5.2.0.

Issue Number: #27

## What is the new behavior?

Code-Style contains the configurations for all the new available versions of `codelyzer`.

New rules have been added in 5.0.x:
  - `"prefer-on-push-component-change-detection": false`
  - `"relative-url-prefix": true`
  - `"template-accessibility-alt-text": true`
  - `"template-accessibility-elements-content": true`
  - `"template-accessibility-label-for": true`
  - `"template-accessibility-tabindex-no-positive": true`
  - `"template-accessibility-table-scope": true`
  - `"template-accessibility-valid-aria": true`
  - `"template-click-events-have-key-events": true`
  - `"template-mouse-events-have-key-events": true`
  - `"template-no-any": true`
  - `"template-no-autofocus": true`
  - `"template-no-call-expression": false`
  - `"template-no-distracting-elements": true`

New rule has been added in 5.1.x:
  - `"use-injectable-provided-in": false`

The effects, by "@ngrx/effects", needs to be `@Injectable()` also but cannot be "provided" in the "root". For this reason, we should not force the usage of `{ providedIn: "root" }` when declaring `Injectable`

## Does this PR introduce a breaking change?

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: lib/codelyzer/5.0.x/tslint.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    "template-accessibility-elements-content": true,
+    "template-accessibility-label-for": true,
+    "template-accessibility-tabindex-no-positive": true,
+    "template-accessibility-table-scope": true,
+    "template-accessibility-valid-aria": true,
+    "template-banana-in-box": true,
+    "template-click-events-have-key-events": true,
+    "template-conditional-complexity": [true, 4],
+    "template-cyclomatic-complexity": [true, 10],
+    "template-mouse-events-have-key-events": true,
```
Comment (christophercr) [inline lib/codelyzer/5.0.x/tslint.json]: Question: are we sure we want to enforce this? As I can see, [this requires more action handlers to be used](https://github.com/mgechev/codelyzer/blob/8e6185e8bb1165ce57d57d742145d901f80fef52/test/templateMouseEventsHaveKeyEventsRule.spec.ts#L52)

File: lib/codelyzer/5.0.x/tslint.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    "prefer-on-push-component-change-detection": false,
+    "relative-url-prefix": true,
+    "template-accessibility-alt-text": true,
+    "template-accessibility-elements-content": true,
+    "template-accessibility-label-for": true,
+    "template-accessibility-tabindex-no-positive": true,
+    "template-accessibility-table-scope": true,
+    "template-accessibility-valid-aria": true,
+    "template-banana-in-box": true,
+    "template-click-events-have-key-events": true,
```
Comment (christophercr) [inline lib/codelyzer/5.0.x/tslint.json]: Same question: are we sure we want to enforce this? [This requires more action handlers to be used](https://github.com/mgechev/codelyzer/blob/8e6185e8bb1165ce57d57d742145d901f80fef52/test/templateClickEventsHaveKeyEventsRule.spec.ts#L165)

File: lib/codelyzer/5.1.x/tslint.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    "template-cyclomatic-complexity": [true, 10],
+    "template-mouse-events-have-key-events": true,
+    "template-no-any": true,
+    "template-no-autofocus": true,
+    "template-no-call-expression": false,
+    "template-no-distracting-elements": true,
+    "template-no-negated-async": true,
+    "template-use-track-by-function": true,
+    "use-component-view-encapsulation": true,
+    "use-injectable-provided-in": false,
```
Comment (christophercr) [inline lib/codelyzer/5.1.x/tslint.json]: I think we should remove it if is disabled anyway right? Same for `5.2.x` config ;)

**Comment by SuperITMan** [comment]
@christophercr 
About the disabled rules, I'll remove them 👍 

Indeed, codelyzer seems to be compatible with all the angular versions. See their peer dependencies:
```json
"peerDependencies": {
    "@angular/compiler": ">=2.3.1 <10.0.0 || >9.0.0-beta <10.0.0 || >9.1.0-beta <10.0.0 || >9.2.0-beta <10.0.0",
    "@angular/core": ">=2.3.1 <10.0.0 || >9.0.0-beta <10.0.0 || >9.1.0-beta <10.0.0 || >9.2.0-beta <10.0.0",
    "tslint": "^5.0.0"
  }
```

About the rules `"template-mouse-events-have-key-events": true` and `"template-click-events-have-key-events": true,` if you use angular material components and you add a `(click)=""` event on the component, the key events are automatically added by Angular.

I think we should enforce the framework to be usable by everyone.
In the case of Stark, we won't have to adapt too much code (2 components should be adapted actually). What do you think ?

Or maybe should we enforce this rule only in the framework Stark ? 😕

**Comment by christophercr** [comment]
> if you use angular material components and you add a (click)="" event on the component, the key events are automatically added by Angular.

I didn't know that... but I cannot find any description of that behavior in the Angular Material docs 🤔 

> I think we should enforce the framework to be usable by everyone.

I understand that this is a "recommended" guideline for accessibility and I don't have a strong opinion about it. So to me it depends on how this will impact the current users of code-style at NBB.

**Comment by SuperITMan** [comment]
To be honnest, I didn't know either. I just discovered it by playing a bit. See the following example on stackblitz: https://stackblitz.com/edit/angular-jthy3s?file=src%2Fapp%2Fapp.component.html 
```html
<button (click)="click()">This is a button</button>

<button (click)="click()" mat-button mat-raised-button>This is a button</button>
```

The first button will generate this:
image

The second one (the material one):
image

But I don't know which logic is behind this...

About the click event, you're probably right. It's maybe too strict since it will have an impact on every `<div (click)="click()"></div>`, `<span (click)="click()"></span>`, ...

I suggest so we set those 2 rules in Stark but not in code-style. What do you think @nicanac ?

**Comment by SuperITMan** [comment]
After talking with @nicanac we decided to set the rules `"template-mouse-events-have-key-events": true` and `"template-click-events-have-key-events": true` only in Stark for now. 
I removed them from this PR 😊 
@nicanac @christophercr could you please have a look ?

**Pull request merged**